src/examples/poi-crosschecker/poi-crosschecker.ts

```diff
@@ -82,23 +77,42 @@ const run = async () => {
         `\n📮 A new message has been received! Parse, validate, and store\n`
           .green
       );
-      const message = observer.readMessage(msg, NPOIMessage);
-      const attestation: Attestation = await observer.prepareAttestation(
-        message,
-        NPOIMessage
+      const message = await observer.readMessage({
+        msg,
+        topic,
+        types: [
```

Comment:
Would it be nice to declare `types` as a constant? it is used in L144 as well. 
Not sure if it would be worth it to try to generate `types` from the `radioPayload` object 🤔

Reply:
I also thought about generating types, but it will be a bit tricky since we're still dealing with JSON 😞 Maybe in the MVP stage we can try to automate this, but for now I think it's ok to leave the type specification to be done on the Radio side manually. That's the approach in the [Eth spec for message signing](https://eips.ethereum.org/EIPS/eip-712) so I think we're good 😁 I'll definitely extract it as a constant though.

Reply:
Moved it to a constant, great suggestion 🚀 I also thought about generating the types, but it seems really hard to do that in TS since the types are stripped away during runtime, so even if we were to dynamically get them for `writeMessage` using something like `typeof` on all the values, we couldn't use the same logic in `readMessage`, since there we just get a vague object. We can try to do it again during the MVP stage though.

Also, [EIP-712](https://eips.ethereum.org/EIPS/eip-712) uses the pattern we have right now, so it's not too bad 😅